config/src/main/java/org/springframework/security/config/annotation/web/configurers/oauth2/server/resource/OAuth2ResourceServerConfigurer.java

```diff
@@ -272,6 +300,9 @@ public void configure(H http) {
 		BearerTokenAuthenticationFilter filter = new BearerTokenAuthenticationFilter(resolver);
 		filter.setBearerTokenResolver(bearerTokenResolver);
 		filter.setAuthenticationEntryPoint(this.authenticationEntryPoint);
```

Comment:
Since `setAuthenticationEntryPoint` is deprecated, I think it would be best to no longer call it.

What if the code instead did `filter.setAuthenticationFailureHandler(getAuthenticationFailureHandler())` where `getAuthenticationFailureHandler` is something like (pseudocode):

```java
if (authenticationFailureHandler != null) {
    return;
}
return new Adapter(authenticationEntryPoint);
```

Reply:
I agree that it'd be better, but I think we risk changing behavior. Notably because `BearerTokenAuthenticationFilter#doFilterInternal` explicitly calls `authenticationEntryPoint.commence(...)`, without using the `FailureHandler` yet, with this code:

```java
	protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain)
			throws ServletException, IOException {
		String token;
		try {
			token = this.bearerTokenResolver.resolve(request);
		}
		catch (OAuth2AuthenticationException invalid) {
			this.logger.trace("Sending to authentication entry point since failed to resolve bearer token", invalid);
			this.authenticationEntryPoint.commence(request, response, invalid);
			return;
		}
		// ...
	}
```

So for the following FilterChain, the behavior would be different:

```
AuthenticationEntryPoint someEntryPoint = getEntryPoint();
http
	.authorizeRequests()
		.anyRequest().authenticated()
		.and()
	.oauth2ResourceServer()
		.authenticationEntryPoint(authenticationEntryPoint())
		.jwt();
```

Reply:
I also noticed that I marked `OAuth2ResourceServerConfigurer#authenticationEntryPoint` as deprecated, but the Entry Point is also used in `OAuth2ResourceServerConfigurer#registerDefaultEntryPoint`, so that deprecation is incorrect, we won't be able to "just" remove the field in 6.0 .

I'm unsure what the best course of action is.

Reply:
I see. Sorry, I had missed that `BearerTokenAuthenticationFilter` uses `this.authenticationEntryPoint` directly.

Since `BearerTokenAuthenticationFilter` continues to use `authenticationEntryPoint`, then it can't be deprecated at this point. That means waiting on it using the new adapter as well, at least until there is time to refactor `BearerTokenAuthenticationFilter`.

At this point, I think it's best to remove the `BearerTokenAuthenticationFilter` work from this PR. I think that in down the road [the DSL will change to use `AuthenticationFilter`](https://github.com/spring-projects/spring-security/issues/11983), which would render the deprecation of `BearerTokenAuthenticationFilter#setAuthenticationEntryPoint` moot.